libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.ts

```diff
@@ -30,15 +48,101 @@ export class ChartJSService {
+    const suffix = datalabelOptions?.valueSuffix ? datalabelOptions.valueSuffix : '';
+
+    // We need to modify the datasets in order to add tooltips.
+    const datasets =
+      datalabelOptions?.showCurrent || datalabelOptions?.showMax || datalabelOptions?.showMin
+        ? this.createDatasets(this.addTooltips(data, datalabelOptions, suffix), highlightedElements)
+        : this.createDatasets(data, highlightedElements);
+
     const options = this.createOptionsObject({ type, customOptions, annotations });
-    const config = this.createConfigurationObject(type, datasets, options, dataLabels);
+    let config = this.createConfigurationObject(type, datasets, options, dataLabels);
+
+    if (type === ChartTypes.stock) {
+      config.options.plugins.tooltip.callbacks.label = (context) => {
+        return context.formattedValue + suffix;
+      };
+
+      config.options.scales.y.ticks.callback = (value) => {
+        return value + suffix;
+      };
+
+      // assuming that the first dataset controls the datespan.
+      const chartPeriod = this.chartConfigService.findChartPeriod(datasets[0]);
+      config.options = this.handleLocalization(
+        config.options,
+        chartPeriod,
+        datalabelOptions.locale
+      );
+    }
+
     this.initializeNewChart(targetElement.nativeElement, config);
   }
 
+  private getLocale(key: string) {
+    return { 'da-DK': da, 'en-US': enUS }[key];
+  }
+  private getDisplayFormats(key: string) {
+    return {
+      'da-DK': {
+        hour: 'HH:mm',
+        day: 'd MMM',
+      },
+      'en-US': {
+        hour: 'HH:mm',
+        day: 'MMM d',
+      },
+    }[key];
+  }
+
+  private handleLocalization(options, chartPeriod, language): ChartOptions {
+    language = language || 'en-US';
+    // Handle localization in graph.
+    options.locale = language;
+
+    // Update chart options with the given period.
+    const scaleX = options.scales.x as any;
+    scaleX.time.unit = chartPeriod;
+    scaleX.adapters = {
+      date: {
+        locale: this.getLocale(language),
+      },
+    };
+    scaleX.time.displayFormats = this.getDisplayFormats(language);
+
+    let tooltipDateformat = '';
+    switch (chartPeriod) {
+      case dataDateSpan.oneDay:
+        tooltipDateformat = 'HH:mm';
+        break;
+      case dataDateSpan.oneWeek:
+      case dataDateSpan.oneMonth:
+      case dataDateSpan.threeMonths:
+      case dataDateSpan.sixMonths:
+      case dataDateSpan.oneMonth:
+      case dataDateSpan.oneYear:
+        tooltipDateformat = this.getDisplayFormats(language).day;
+        break;
+      case dataDateSpan.fiveYears:
+        tooltipDateformat = 'LLL yy';
+        break;
+    }
+
+    options.plugins.tooltip.callbacks.title = (tooltipItems) => {
+      const date = toDate(tooltipItems[0]?.parsed?.x);
+      if (date.valueOf()) {
+        return format(date, tooltipDateformat, { locale: this.getLocale(language) });
+      }
+    };
+
+    return options;
+  }
```

Comment:
I can't seem to find anything at the moment - but I need to check up on if there's a neat place to store this in the library. 

Alternatively we could consider moving it to a separate `kirby-chart-i18n-service` or something like that.